Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGppProtobufExtensions.cs

```diff
@@ -0,0 +1,610 @@
+                    ReportStatus = processDetouringStatus.ReportStatus,
+                    ProcessName = processDetouringStatus.ProcessName,
+                    StartApplicationName = processDetouringStatus.StartApplicationName,
+                    StartCommandLine = processDetouringStatus.StartCommandLine,
+                    NeedsInjection = processDetouringStatus.NeedsInjection,
+                    Job = processDetouringStatus.Job,
+                    DisableDetours = processDetouringStatus.DisableDetours,
+                    CreationFlags = processDetouringStatus.CreationFlags,
+                    Detoured = processDetouringStatus.Detoured,
+                    Error = processDetouringStatus.Error
+                });
+            }
+
+            return processExecutionMonitoringReportedEvent;
+        }
+
+        /// <nodoc />
+        public static ProcessFingerprintComputationEvent_XLGpp ToProcessFingerprintComputationEvent_XLGpp(this ProcessFingerprintComputationEventData data, uint workerID, PathTable pathTable)
+        {
+            var Uuid = Guid.NewGuid().ToString();
+
+            var processFingerprintComputationEvent = new ProcessFingerprintComputationEvent_XLGpp
+            {
+                UUID = Uuid,
+                WorkerID = workerID,
+                Kind = (FingerprintComputationKind_XLGpp)data.Kind,
+                PipID = data.PipId.Value,
+                WeakFingerprint = new WeakContentFingerPrint_XLGpp()
+                {
+                    Hash = data.WeakFingerprint.Hash.ToFingerprint_XLGpp()
+                },
+            };
+
+            foreach (var strongFingerprintComputation in data.StrongFingerprintComputations)
+            {
+                var processStrongFingerprintComputationData = new ProcessStrongFingerprintComputationData_XLGpp()
+                {
+                    PathSet = strongFingerprintComputation.PathSet.ToObservedPathSet_XLGpp(pathTable),
+                    PathSetHash = new ContentHash_XLGpp()
+                    {
+                        Value = strongFingerprintComputation.PathSetHash.ToString()
+                    },
+                    UnsafeOptions = strongFingerprintComputation.UnsafeOptions.ToUnsafeOptions_XLGpp(),
+                    Succeeded = strongFingerprintComputation.Succeeded,
+                    IsStrongFingerprintHit = strongFingerprintComputation.IsStrongFingerprintHit,
+                    ComputedStrongFingerprint = new StrongContentFingerPrint_XLGpp()
+                    {
+                        Hash = strongFingerprintComputation.ComputedStrongFingerprint.Hash.ToFingerprint_XLGpp()
+                    }
+                };
+
+                foreach (var pathEntry in strongFingerprintComputation.PathEntries)
+                {
+                    processStrongFingerprintComputationData.PathEntries.Add(pathEntry.ToObservedPathEntry_XLGpp(pathTable));
+                }

```

Comment:
nit: I'd replace these small foreach loops with something lie this:
`processStrongFingerprintComputationData.PathEntries.AddRange(strongFingerprintComputation.PathEntries.Select(pathEntry => pathEntry.ToObservedPathEntry_XLGpp(pathTable)))`